### PR TITLE
Update npm deps

### PR DESCRIPTION
**Pull request by lnicola**

### CODE REVIEW

**Comment by lnicola** [comment]
r? @matklad

I didn't update `@rollup/plugin-commonjs` all the way because of a strange error:

    The language client requires VS Code version ^1.53.0 but received version 1.56.2.

We might as well switch to `esbuild`, which seems to be the most liked bundler these days /s.

File: editors/code/.vscodeignore [hunk 1]
```diff
@@ -1,5 +1,8 @@
 **
 !language-configuration.json
+!out
+out/**
+!out/src
 !out/src/main.js
```
Comment (matklad) [inline editors/code/.vscodeignore:6]: So what exactly are we pulling in from `out`?

Comment (bjorn3) [reply editors/code/.vscodeignore:6]: The `out/src` dir. By the way `!out/src/main.js` is not necessary if you have `!out/src` but not `out/src/**`.

**Comment by lnicola** [comment]
Status: this is marked as a draft because it breaks loading the extension. I didn't notice it because there's no error thrown.

**Comment by matklad** [comment]
Scheduled for r+ on Monday!

**Comment by yaymukund** [comment]
> r? @matklad
> 
> I didn't update `@rollup/plugin-commonjs` all the way because of a strange error:
> 
> ```
> The language client requires VS Code version ^1.53.0 but received version 1.56.2.
> ```
> 
> We might as well switch to `esbuild`, which seems to be the most liked bundler these days /s.
Very mysterious bug.

It's worth noting that this blocks updating rollup itself. If you update rollup to 2.55 without commonjs 19, you run into https://github.com/rollup/rollup/issues/4195 . I ran into this while trying to package this extension for Nix.

Sorry to ping you on this old issue, I just thought it might save you some debugging in the future because it was fairly tricky to chase down.

**Comment by lnicola** [comment]
Thanks for the ping, though I'll probably forget by then.

---

Anyway, as a non-web developer, the whole JS bundler ecosystem seems Kafkaesque.

**Comment by matklad** [comment]
One morning, as Gregor Samsa was waking up from anxious dreams, he discovered that his bed had been moved into `node_modules`.

**Comment by matklad** [comment]
Curiously, the docs now put esbuild to the first place: 

https://code.visualstudio.com/api/working-with-extensions/bundling-extension#using-esbuild

I wonder if we should switch to that? On the one hand, I wouldn't want to pull yet another ecosystem (esbuild is in go), on the other hand, the pain with rollup/webpack is real, and the example in the docs does seem nice. 

Fun fact: esbuild could have been written in Rust, but there was [this blockin issue](https://github.com/rust-lang/rust/issues/34283).

**Comment by yaymukund** [comment]
> One morning, as Gregor Samsa was waking up from anxious dreams, he discovered that his bed had been moved into `node_modules`.

terrifying! :joy: 

I can try to integrate esbuild in the coming week, if this is something you feel is worth pursuing. 

Also, that is a neat bit of trivia about esbuild. I wonder if swc has ever run into the same thing.

**Comment by lnicola** [comment]
> I can try to integrate esbuild in the coming week, if this is something you feel is worth pursuing.

If it means less npm deps in `package.json`, yes!

> `npm i --save-dev esbuild`

Oh.

**Comment by yaymukund** [comment]
> If it means less npm deps in package.json, yes!

It would still reduce the number of dependencies because [it has no dependencies of its own](https://www.npmjs.com/package/esbuild?activeTab=dependencies). 

> I wonder if we should switch to that? On the one hand, I wouldn't want to pull yet another ecosystem (esbuild is in go), on the other hand, the pain with rollup/webpack is real, and the example in the docs does seem nice.

For what it's worth, it wouldn't require Go as a build dependency. `npm` just pulls the static binary from the registry.

**Pull request merged**